src/reason_parser.mly

```diff
@@ -782,6 +782,21 @@ let rec string_of_longident = function
 
 let built_in_explicit_arity_constructors = ["Some"; "Assert_failure"; "Match_failure"]
 
+(* Replace dot with bangdot
+   e.g.: "Int32.rem" ~> "Int32!.rem" *)
+let prepare_name_for_call name =
+  if String.contains name '.' then
+    let rec replace str n = match str with
+        | "" -> ""
+        | str ->
+          let ch = (String.get str 0) in
+          let new_str = if Char.compare ch '.' == 0 then "!." else String.make 1 ch in
```

Comment:
Not very familiar with the logic with Longident.parse -- what's the magic here to replace "." with "!."?